sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LocalRelation.scala

```diff
@@ -80,7 +80,7 @@ case class LocalRelation(
 
   override def computeStats(): Statistics =
     Statistics(sizeInBytes = EstimationUtils.getSizePerRow(output) * data.length,
-      rowCount = Some(data.size))
+      rowCount = Some(data.length))
```

Comment:
Actually, this was mentioned by @HyukjinKwon first. I thought it's addressed.
- https://github.com/apache/spark/pull/41064#discussion_r1186956202